### PR TITLE
Master issue57 yaml

### PR DESCRIPTION
**Pull request by hugo6**

### CODE REVIEW

**Comment by emuus** [comment]
ok for me

File: src/hammr/commands/template/template.py [hunk 1]
```diff
@@ -293,11 +294,11 @@ def do_create(self, args):
                 #delete tmp dir
                 shutil.rmtree(constants.TMP_WORKING_DIR)
             os.mkdir(constants.TMP_WORKING_DIR)
-            file = open(constants.TMP_WORKING_DIR + os.sep + constants.TEMPLATE_JSON_NEW_FILE_NAME, "w")
```
Comment (lqueiroga) [inline src/hammr/commands/template/template.py]: - We need to handle both YAML and JSON
- We need to create a dedicated function for this

File: src/hammr/commands/template/template.py [hunk 2]
```diff
@@ -357,12 +358,12 @@ def do_build(self, args):
...
 
             #--
-            template=validate_json_file(doArgs.file)
+            template=check_extension_and_validate(doArgs.file)
             if template is None:
                 return 2
 
             if doArgs.id:
-                myAppliance = self.api.Users(self.login).Appliances(doArgs.id).Get()
+                myAppliance = self.api.Users(self.login).Appliances().Getall(Query="dbId=="+doArgs.id)
```
Comment (lqueiroga) [inline src/hammr/commands/template/template.py]: Need to test with old code

File: src/hammr/hammr.py [hunk 3]
```diff
@@ -188,19 +184,17 @@ def check_credfile(credfile):
     password=mainArgs.password
     sslAutosigned=True
 else:
-    credfile="credentials.json"
```
Comment (lqueiroga) [inline src/hammr/hammr.py]: We need to handle these 2 extensions

File: src/hammr/utils/constants.py [hunk 4]
```diff
@@ -24,8 +24,9 @@
 TMP_WORKING_DIR=tempfile.gettempdir() + os.sep + "hammr-" + str(os.getpid())
 HTTP_TIMEOUT=30
 
-TEMPLATE_JSON_FILE_NAME="template.json"
-TEMPLATE_JSON_NEW_FILE_NAME="template.json"
```
Comment (lqueiroga) [inline src/hammr/utils/constants.py]: We need to handle these 2 extensions

File: src/hammr/utils/hammr_utils.py [hunk 5]
```diff
@@ -96,9 +96,37 @@ def check_mandatory_create_account(iterables, type):
...
+    fileExtension = os.path.splitext(file)[1]
+    if fileExtension == ".yml":
+        print "you provided a yaml file, checking the syntax..."
+        data = generics_utils.check_yaml_syntax(file)
+    elif fileExtension == ".json":
+        print "you provided a json file, checking the syntax..."
         data = generics_utils.check_json_syntax(file)
+    else:
+        printer.out("please provide a json or yaml file \n", printer.ERROR)
+        exit(1)
```
Comment (lqueiroga) [inline src/hammr/utils/hammr_utils.py]: raise an exception instead of exit

File: src/hammr/utils/hammr_utils.py [hunk 6]
```diff
@@ -96,9 +96,37 @@ def check_mandatory_create_account(iterables, type):
 
     return iterables
 
-def validate_json_file(file):
-    try:
+def check_extension_and_load_data(file):
```
Comment (lqueiroga) [inline src/hammr/utils/hammr_utils.py]: rename this method to load_data()

File: src/hammr/utils/hammr_utils.py [hunk 7]
```diff
@@ -96,9 +96,37 @@ def check_mandatory_create_account(iterables, type):
...
+    fileExtension = os.path.splitext(file)[1]
+    if fileExtension == ".yml":
+        print "you provided a yaml file, checking..."
+        template = validate_configurations_file(file, isJson=False)
+    elif fileExtension == ".json":
+        print "you provided a json file, checking..."
+        template = validate_configurations_file(file, isJson=True)
+    else:
+        printer.out("please provide a json or yaml file \n", printer.ERROR)
+        exit(1)
```
Comment (lqueiroga) [inline src/hammr/utils/hammr_utils.py]: raise an exception instead of exit

**Comment by lqueiroga** [comment]
I closed this PR because I worked on it and send a new PR with all the code done here (see PR 149)

**Pull request closed without merging**